components/raftstore/src/store/fsm/store.rs

```diff
@@ -2035,11 +2035,20 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
 
         let mut is_overlapped = false;
         let mut regions_to_destroy = vec![];
-        for (_, id) in meta.region_ranges.range((
+        for (key, id) in meta.region_ranges.range((
             Excluded(data_key(msg.get_start_key())),
             Unbounded::<Vec<u8>>,
         )) {
-            let exist_region = &meta.regions[id];
+            let exist_region = match meta.regions.get(id) {
```

Comment:
We have many other places using `meta.regions[..]`. How about extracting to a function for it?

Reply:
Thought about it, but those places were less frequently visited and context everywhere might be too verbose.